### ISSUE TITLE
[Feature Request] Publish Mojo and Mojo: Nightly .vsix extensions to Open VSX

### ISSUE THREAD

**Issue by jthomasmock** [modular/modular#2033]
### Review Mojo's priorities

- [X] I have read the [roadmap and priorities](https://docs.modular.com/mojo/roadmap.html#overall-priorities) and I believe this request falls within the priorities.

### What is your request?

Howdy Modular team!

It would be awesome if y'all could cross-publish the Mojo and Mojo Nightly extensions to the Open VSX marketplace.

You can read on about Open VSX at the Eclipse Foundation's [Open VSX FAQ](https://www.eclipse.org/legal/open-vsx-registry-faq/).

## How can you publish to Open VSX?

The docs to publish an extension can be found [here](https://github.com/eclipse/openvsx/wiki/Publishing-Extensions). This process is straightforward and shouldn't take too long. Essentially, you need an authentication token and to execute the `ovsx publish` command to publish your extension. There's also [a doc](https://github.com/open-vsx/publish-extensions/blob/master/docs/direct_publish_setup.md) explaining the whole process with an example GitHub Action workflow.

## Context

Unfortunately, as Microsoft prohibits usages of the Microsoft marketplace by any other products or redistribution of `.vsix` files from it, in order to use VS Code extensions in non-Microsoft products, we kindly ask that you take ownership of the VS Code extension namespace in [Open VSX](https://open-vsx.org/) and publish this extension on Open VSX.

## What is Open VSX? Why does it exist?

Open VSX is a vendor neutral alternative to the MS marketplace used by most other derivatives of VS Code like [VSCodium](https://vscodium.com/), [Gitpod](https://www.gitpod.io), [OpenVSCode](https://github.com/gitpod-io/openvscode-server), [Theia](https://theia-ide.org/)-based IDEs, and so on.

**Comment by JoeLoser** [comment]
@River707 @modocache do you guys know much about this?

**Comment by River707** [comment]
Yes, this is something we know about. There have been a lot of moving parts for Mojo releases, but this is something we want to do at some point in the future, just figuring out the timing. Really glad when people ask about this!

**Comment by jthomasmock** [comment]
Thanks for the update! We look forward to the cross-publish and thanks for all the very exciting work.

**Comment by delta-atk** [comment]
It would be very nice to see this, as the Download buttons from the Version History on the marketplace site seem to be (permanently?) gone now ([modular-mojotools.vscode-mojo](https://marketplace.visualstudio.com/items?itemName=modular-mojotools.vscode-mojo)).
This means it is really hard now to get the extension for Code OSS / Codium or similar, which is very sad.

**Comment by incansvl** [comment]
Apologies for the potentially dumb questions but-

1. Is this the reason the official Mojo extension is not found if it is searched for from vscodium (on Linux)?
2. Is there a workaround to get the extension installed via another route? (e.g. manual download)?

Tough to learn a new language without decent editing and debug tools.

**Comment by benz0li** [comment]
There is https://open-vsx.org/extension/modular-mojotools/vscode-mojo and https://open-vsx.org/extension/modular-mojotools/vscode-mojo-nightly.

Published by @zbowling about a month ago. Maybe you can pick up on his initial work.

**Comment by benz0li** [comment]
> - We've released a new version of the Mojo extension for Visual Studio Code that now works with **both** the stable and nightly Mojo releases. You can install the Mojo extension from either the [Visual Studio Code Marketplace](https://marketplace.visualstudio.com/items?itemName=modular-mojotools.vscode-mojo) or the [Open VSX Registry](https://open-vsx.org/extension/modular-mojotools/vscode-mojo).

– https://docs.modular.com/mojo/changelog/#25-6-highlights

ℹ️ The latest version of the [Mojo extension in the Open VSX Registry](https://open-vsx.org/extension/modular-mojotools/vscode-mojo) is 25.4.0 – published end of June 2025.

**Comment by AmaranthineCodices** [comment]
> > * We've released a new version of the Mojo extension for Visual Studio Code that now works with **both** the stable and nightly Mojo releases. You can install the Mojo extension from either the [Visual Studio Code Marketplace](https://marketplace.visualstudio.com/items?itemName=modular-mojotools.vscode-mojo) or the [Open VSX Registry](https://open-vsx.org/extension/modular-mojotools/vscode-mojo).
> 
> – https://docs.modular.com/mojo/changelog/#25-6-highlights
> 
> ℹ️ The latest version of the [Mojo extension in the Open VSX Registry](https://open-vsx.org/extension/modular-mojotools/vscode-mojo) is 25.4.0 – published end of June 2025.

This is now fixed, we had some issues with the initial rollout that took some time to resolve.
